internal/temporalcli/commands.extension.go

```diff
@@ -159,21 +162,110 @@ func lookupExtension(cmdPrefix, extArgs []string) (string, []string) {
 		if !isPosArg(arg) {
 			break
 		}
-		// Dashes are converted to underscores so "foo bar-baz" finds "temporal-foo-bar_baz".
-		posArgs = append(posArgs, strings.ReplaceAll(arg, extensionSeparator, argDashReplacement))
+		posArgs = append(posArgs, arg)
 	}
 
 	// Try most-specific to least-specific.
 	parts := append(cmdPrefix, posArgs...)
 	for n := len(parts); n > len(cmdPrefix); n-- {
-		path, err := exec.LookPath(extensionPrefix + strings.Join(parts[:n], extensionSeparator))
```

Comment:
This was finding files on Windows that weren't actually executables; the new `isExecutable` addresses that.